### PR TITLE
cpu/stm32: Cleanup periph_eth

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Cleanup functions `_rw_phy()`, `_phy_read`, and `_phy_write` and rename them to `_mii_reg_{access,read,write}()`.

### Testing procedure

Those functions are needed to during device initialization, so being able to still ping the Nucleo should be testing enough.

### Issues/PRs references

None

### CODE REVIEW

File: cpu/stm32/periph/eth.c [hunk 1]
```diff
@@ -104,40 +104,53 @@ netdev_t *_netdev;
...
 {
     unsigned tmp;
 
     while (ETH->MACMIIAR & ETH_MACMIIAR_MB) {}
     DEBUG("[stm32_eth] rw_phy %x (%x): %x\n", addr, reg, value);
 
-    tmp = (ETH->MACMIIAR & ETH_MACMIIAR_CR) | ETH_MACMIIAR_MB;
-    tmp |= (((addr & 0x1f) << 11) | ((reg & 0x1f) << 6));
-    tmp |= (value >> 16);
+    tmp = CLOCK_RANGE | ETH_MACMIIAR_MB | (((addr & 0x1f) << 11)
```
Comment (maribu) [inline cpu/stm32/periph/eth.c:127]: This previously read the current clock setting from the configuration register. But we now that value at compile time (it is `CLOCK_RANGE`), so we can avoid a read from the peripheral here.

File: cpu/stm32/periph/eth.c [hunk 2]
```diff
@@ -104,40 +104,53 @@ netdev_t *_netdev;
...
+          | ((reg & 0x1f) << 6));
+    if (write) {
+        tmp |= ETH_MACMIIAR_MW;
+        ETH->MACMIIDR = value;
+    }
```
Comment (maribu) [inline cpu/stm32/periph/eth.c:132]: Only writing to the peripheral's data register when actually transferring data to it safes another memory transfer. We also do no longer need the `& 0xffff` by just using `uint16_t` as argument type.

**Comment by maribu** [comment]
ping still works from Nucleo to Linux:

```
2020-10-08 10:39:48,925 #  ping6 fe80::xxx
2020-10-08 10:39:48,932 # 12 bytes from fe80::xxx%5: icmp_seq=0 ttl=64 time=0.427 ms
2020-10-08 10:39:49,933 # 12 bytes from fe80::xxx%5: icmp_seq=1 ttl=64 time=0.418 ms
2020-10-08 10:39:50,932 # 12 bytes from fe80::xxx%5: icmp_seq=2 ttl=64 time=0.423 ms
2020-10-08 10:39:50,932 # 
2020-10-08 10:39:50,936 # --- fe80::xxx PING statistics ---
2020-10-08 10:39:50,941 # 3 packets transmitted, 3 packets received, 0% packet loss
```

And the other way round

```
$ ping -I eth0 fe80::2012:14ff:fe0a:1345
ping: Warning: source address might be selected on device other than: eth0
PING fe80::2012:14ff:fe0a:1345(fe80::2012:14ff:fe0a:1345) from :: eth0: 56 data bytes
64 bytes from fe80::2012:14ff:fe0a:1345%eth0: icmp_seq=1 ttl=64 time=0.456 ms
64 bytes from fe80::2012:14ff:fe0a:1345%eth0: icmp_seq=2 ttl=64 time=0.485 ms
64 bytes from fe80::2012:14ff:fe0a:1345%eth0: icmp_seq=3 ttl=64 time=0.441 ms
64 bytes from fe80::2012:14ff:fe0a:1345%eth0: icmp_seq=4 ttl=64 time=0.424 ms
^C
--- fe80::2012:14ff:fe0a:1345 ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3033ms
rtt min/avg/max/mdev = 0.424/0.451/0.485/0.022 ms
```

File: cpu/stm32/periph/eth.c [hunk 3]
```diff
@@ -104,40 +104,53 @@ netdev_t *_netdev;
...
     unsigned tmp;
 
     while (ETH->MACMIIAR & ETH_MACMIIAR_MB) {}
     DEBUG("[stm32_eth] rw_phy %x (%x): %x\n", addr, reg, value);
 
-    tmp = (ETH->MACMIIAR & ETH_MACMIIAR_CR) | ETH_MACMIIAR_MB;
-    tmp |= (((addr & 0x1f) << 11) | ((reg & 0x1f) << 6));
-    tmp |= (value >> 16);
+    tmp = CLOCK_RANGE | ETH_MACMIIAR_MB | (((addr & 0x1f) << 11)
+          | ((reg & 0x1f) << 6));
```
Comment (benpicco) [inline cpu/stm32/periph/eth.c:128]:
```suggestion
    tmp = CLOCK_RANGE | ETH_MACMIIAR_MB | (((addr & 0x1f) << 11)
        | ((reg & 0x1f) << 6));

```

Small whitespace tweaks to make it even more pleasant to the eye ;)

**Review by benpicco — APPROVED** [review]
Looks good to me.
I think separating the if block with a newline helps to avoid clutter.

Squash if you agree.

**Comment by maribu** [comment]
> I think separating the if block with a newline helps to avoid clutter.
> 
> Squash if you agree.

Done

**Comment by maribu** [comment]
I added another commit with style fixes, so that travis is hopefully happy now.

**Comment by maribu** [comment]
Thanks :-)

**Pull request merged**